test/arcade/patch/agent/cell/PatchCellCARTCD4Test.java

```diff
@@ -0,0 +1,454 @@
+        cellMock.processes.put(Domain.INFLAMMATION, mock(PatchProcessInflammation.class));
+        PatchModule module = mock(PatchModule.class);
+        doAnswer(
+                        invocationOnMock -> {
+                            cellMock.state = invocationOnMock.getArgument(0);
+                            cellMock.module = module;
+                            return null;
+                        })
+                .when(cellMock)
+                .setState(any(State.class));
+        doReturn(new PatchCellTissue(container, locationMock, parametersMock))
+                .when(cellMock)
+                .bindTarget(
+                        any(Simulation.class),
+                        any(PatchLocation.class),
+                        any(MersenneTwisterFast.class));
+        Field div = PatchCell.class.getDeclaredField("divisions");
+        div.setAccessible(true);
+        div.set(cellMock, cellMock.divisionPotential);
+        when(simMock.random.nextDouble()).thenReturn(0.49);
+
+        cellMock.step(simMock);
+
+        assertTrue(cellMock.getState() == State.SENESCENT);
+        assertEquals(AntigenFlag.UNBOUND, cellMock.getBindingFlag());
+        assertFalse(cellMock.getActivationStatus());
+    }
+
+    @Test
+    public void step_whenBoundToBothAntigenAndSelf_setsStateToAnergic() {
+        when(parametersMock.getDouble("APOPTOSIS_AGE")).thenReturn(120960.0);
+        when(parametersMock.getDouble("ANERGIC_FRACTION")).thenReturn(0.5);
+        when(simMock.random.nextDouble()).thenReturn(0.49);
+        cellMock = spy(new PatchCellCARTCD4(container, locationMock, parametersMock));
+        cellMock.processes.put(Domain.METABOLISM, mock(PatchProcessMetabolism.class));
+        cellMock.processes.put(Domain.SIGNALING, mock(PatchProcessSignaling.class));
+        cellMock.processes.put(Domain.INFLAMMATION, mock(PatchProcessInflammation.class));
+        PatchModule module = mock(PatchModule.class);
+        doAnswer(
+                        invocationOnMock -> {
+                            cellMock.state = invocationOnMock.getArgument(0);
+                            cellMock.module = module;
+                            return null;
+                        })
+                .when(cellMock)
+                .setState(any(State.class));
+        doReturn(new PatchCellTissue(container, locationMock, parametersMock))
+                .when(cellMock)
+                .bindTarget(
+                        any(Simulation.class),
+                        any(PatchLocation.class),
+                        any(MersenneTwisterFast.class));
+        cellMock.setBindingFlag(AntigenFlag.BOUND_ANTIGEN_CELL_RECEPTOR);
+
+        cellMock.step(simMock);
+
+        assertTrue(cellMock.getState() == State.ANERGIC);
+        assertEquals(AntigenFlag.UNBOUND, cellMock.getBindingFlag());
+        assertFalse(cellMock.getActivationStatus());
+    }
+
+    @Test
+    public void step_whenBoundToAntigen_setsStateToStimulatory()
```

Comment:
I am not following this test. The rest of the tests here are far more intuitive, but you're adding a bunch of values here that don't seem relevant to the `step` function. Especially later when you set the binding flag to a BOUND_ANTIGEN. Please reevaluate what is relevant to include in this test.